apps/developer-portal/src/components/user-stores/wizards/general-details-userstore.tsx

```diff
@@ -16,70 +16,59 @@
 * under the License.
 */
 
-import { addAlert } from "@wso2is/core/store";
 import { Field, FormValue, Forms } from "@wso2is/forms";
-import React, { ReactElement, useEffect, useState } from "react";
-import { useDispatch } from "react-redux";
+import React, { ReactElement, useState } from "react";
 import { Button, Divider, Grid, Header, Icon } from "semantic-ui-react";
-import { getAType, testConnection } from "../../../api";
+import { testConnection } from "../../../api";
 import { JDBC } from "../../../constants";
-import { AlertLevels, TestConnection, Type, TypeProperty } from "../../../models";
+import { TestConnection, TypeProperty, UserstoreType } from "../../../models";
 
 /**
- * Prop types of the `ConnectionDetails` component 
+ * Prop types of the `GeneralDetails` component 
  */
-interface ConnectionDetailsPropsInterface {
+interface GeneralDetailsUserstorePropsInterface {
     /**
      * Trigger submit
      */
     submitState: boolean;
     /**
      * Submits values
      */
-    onSubmit: (values: Map<string, FormValue>, type: Type) => void;
+    onSubmit: (values: Map<string, FormValue>) => void;
     /**
      * The saved values
      */
     values: Map<string, FormValue>;
     /**
-     * The type ID chosen by the user
+     * The type of the userstore chosen by the user
      */
-    typeId: string;
+    type: UserstoreType;
+    /**
+     * Connection properties.
+     */
+    connectionProperties: TypeProperty[];
+    /**
+     * Basic properties.
+     */
+    basicProperties: TypeProperty[];
 }
 
 /**
- * This component renders the Connection Details step of the wizard
- * @param {ConnectionDetailsPropsInterface} props
- * @return {Promise<any>}
+ * This component renders the General Details step of the wizard
+ * @param {GeneralDetailsUserstorePropsInterface} props
+ * @returns {Promise<any>}
```

Comment:
Add component type as well.
```suggestion
 * @returns {React. ReactElement}
```